packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -135,20 +143,69 @@ export default Rule.Atomic.of<Page, Element>({
 
         const hasDistinguishingStyle = (context: Context = Context.empty()) =>
           Set.from(
-            linkElements.map((link) =>
+            linkElements.map((link) => {
               // If the link element is distinguishable from at least one
               // non-link element, this is good enough.
               // Note that ACT rules draft requires the link-element to be
               // distinguishable from *all* non-link elements in order to be good.
-              nonLinkElements.some(
+              const hasDistinguishableStyle = nonLinkElements.some(
                 (container) =>
-                  isDistinguishable(container, device, context)(link) ||
+                  Distinguishable.isDistinguishable(
+                    container,
+                    device,
+                    context
+                  )(link) ||
                   (context.isHovered(target) &&
-                    hasDistinguishableCursor(container, device, context)(link))
-              )
-                ? Ok.of(ComputedStyles.from(link, device, target, context))
-                : Err.of(ComputedStyles.from(link, device, target, context))
-            )
+                    Distinguishable.hasDistinguishableCursor(
+                      container,
+                      device,
+                      context
+                    )(link))
+              );
+
+              const distinguishableContrast = Set.from(
+                nonLinkElements.flatMap((container) =>
+                  Sequence.from(
+                    pairwiseContrastCache
+                      .get(device, Cache.empty)
+                      .get(context, Cache.empty)
+                      .get(container, Cache.empty)
+                      .get(link, () =>
+                        Distinguishable.getPairwiseContrast(
+                          container,
+                          link,
+                          device,
+                          context
+                        )
+                      )
+                  )
+                )
+              ).toArray();
```

Comment:
```suggestion
              );
```
And we can remove the `toArray` here (after making `ElementDistinguishable` accept any `Iterable`).
That will make future changes easier, since it is now the `.of` who takes care about what precise format is needed to build the instance, and callers don't care.